pkg/providers/pricing/pricing.go

```diff
@@ -29,62 +43,189 @@ type Provider interface {
 	UpdateSpotPricing(context.Context) error
 }
 
+type pricesStorage = map[string]float64
+
 type DefaultProvider struct {
-	muPriceLastUpdatedTimestamp sync.RWMutex
-	muOnDemand                  sync.RWMutex
-	muSpot                      sync.RWMutex
+	region string
+
+	muOnDemand     sync.RWMutex
+	onDemandPrices pricesStorage
+	muSpot         sync.RWMutex
+	spotPrices     pricesStorage
 }
 
 func NewDefaultProvider(ctx context.Context, region string) *DefaultProvider {
-	// Implement me
+	p := &DefaultProvider{
+		region:         region,
+		onDemandPrices: make(pricesStorage),
+		spotPrices:     make(pricesStorage),
+	}
+
+	if err := p.loadInitialPrices(); err != nil {
+		klog.Errorf("Failed to load initial prices: %v", err)
+	}
+
+	return p
+}
+
+func (p *DefaultProvider) loadInitialPrices() error {
+	p.muOnDemand.Lock()
+	defer p.muOnDemand.Unlock()
+
+	// Get the directory of the current file
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		return fmt.Errorf("failed to get current file path")
+	}
+	dir := filepath.Dir(filename)
```

Comment:
Maybe implement this like alibaba cloud: https://github.com/cloudpilot-ai/karpenter-provider-alibabacloud/blob/e92dee8aa5e8de7c9b2e1268238b8b163a1eb310/pkg/providers/pricing/pricing.go#L173 ?